kube-client/src/client/mod.rs

```diff
@@ -238,15 +237,18 @@ impl Client {
         Ok(text)
     }
 
-    /// Perform a raw HTTP request against the API and get back the response
-    /// as a stream of bytes
-    pub async fn request_text_stream(
-        &self,
-        request: Request<Vec<u8>>,
-    ) -> Result<impl Stream<Item = Result<Bytes>>> {
+    /// Perform a raw HTTP request against the API, get back the response
+    /// and return it as an object that implements [`AsyncBufRead`] and thus
+    /// [`AsyncBufReadExt`]. Users can call `.lines()` on it to get a newline
+    /// buffered stream of logs.
```

Comment:
Another attempt at documentation:

    /// Perform a raw HTTP request against the API and stream the response body.
    ///
    /// The response can be processed using [`AsyncReadExt`](futures::AsyncReadExt) 
    /// and [`AsyncBufReadExt`](futures::AsyncBufReadExt).

I think the `.lines()` is too opinionated at the client (this is generic enough that it wont just be used for logs), but instead the links to the docs what can be used for processing.

Emphasis that this still streams the response.

The link to `AsyncBufReadExt` was broken (it is not imported -> `cargo doc` doesn't know what to link it to).